pkg/cmd/roachtest/versionupgrade.go

```diff
@@ -321,7 +321,11 @@ func uploadAndStartFromCheckpointFixture(nodes nodeListOption, v string) version
 func binaryUpgradeStep(nodes nodeListOption, newVersion string) versionStep {
 	return func(ctx context.Context, t *test, u *versionUpgradeTest) {
 		c := u.c
-		args := u.uploadVersion(ctx, t, nodes, newVersion)
+
+		// NB: We could technically stage the binary on all nodes before
+		// restarting each one, but on Unix it's invalid to write to an
+		// executable file while it is currently running. So we do the
+		// simple thing and upload it serially instead.
```

Comment:
just FYI if you were to implement something like this in the future: upload the binary with a new name in parallel here, then rename it into place in the sequential step below. This parallelizes the time-consuming task and only keeps a short step for the critical path.

Reply:
oo, good idea. I'm hoping to eventually pull together all these various version upgraded related tests into dedicated testing infrastructure for this sort of thing, can address it then.